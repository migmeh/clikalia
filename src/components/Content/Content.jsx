import React from 'react';
import './Content.scss';
import Card from '../Card'
import 'bootstrap/dist/css/bootstrap.css';

export default class Content extends React.Component {
    componentDidMount() {
        const pokemonLength = this.props.all_pokemon.length;
        this.props.fetchPokemon(pokemonLength, 40);

        window.addEventListener("scroll", function(){
            var wrap = document.getElementById('content');
            var contentHeight = wrap.offsetHeight;
            var yOffset = window.pageYOffset;
            var y = yOffset + window.innerHeight;
            if(!this.props.loading_pokemon && (y >= contentHeight))
            {
                this.props.fetchPokemon(this.props.all_pokemon.length, 10);
            }
        }.bind(this))
    }

    renderCard = (pokemon, key) => {
        return <Card {...pokemon} id={key+1}
            key={key+1}
            show_modal={this.props.show_modal}
            showModal={this.props.showModal}
        />
    };

    renderCards = () => {
        const { all_pokemon } = this.props;
        return all_pokemon.map(this.renderCard);
    }

    render() {
        return (




            <div className="contx" id="content">


                <div className="dos cards-table">
                    {this.renderCards()}
                </div>
                { this.props.loading_pokemon?
                    <p className="App-intro"> loading ...</p>
                    : ""
                }

    <div className="s"></div>
            </div>
        );
    }
}
