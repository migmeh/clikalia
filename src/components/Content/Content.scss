$light-gray: #edf3f8;
$gray: #879395;
$green: #1abc9c;
$blue: #2d8abd;
$yellow: #ffc000;

.contx {
  position: relative;

  max-width: 100%;
  float:right;
  .cards-table {
    max-width: 66%;
    float:right;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
  }
}


@media screen and (max-width:600px) and (min-width:0px){

  .cards-table {
    width: 100%;}
}
