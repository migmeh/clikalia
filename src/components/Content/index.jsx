import { connect } from 'react-redux'
import { 
    fetchPokemon, 
} from '../../actions/Pokemon'
import { 
    showModal, 
} from '../../actions/Modal'
import Content from './Content'


const mapStateToProps = (state) => {
    return {
        all_pokemon:     state.pokemon.all_pokemon,
        loading_pokemon: state.pokemon.fetching,
        max_pokemon:     state.pokemon.max_pokemon,
        show_modal:      state.modal.show_modal,
    }
}


const mapDispatchToProps = (dispatch) => {
    return {
        fetchPokemon: (offset, limit) => {
            dispatch(fetchPokemon(offset, limit));
        },
        showModal: (pokemon_name) => {
            dispatch(showModal(pokemon_name));
        }
    }
}


export default connect(mapStateToProps, mapDispatchToProps)(Content);