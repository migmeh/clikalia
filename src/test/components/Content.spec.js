import React from 'react';
import { shallow } from 'enzyme';

import Content from '../../components/Content/Content';

describe('Content', () => {
  it('Should render correctly', () => {
    const tree = shallow(<Content all_pokemon={[]} fetchPokemon={jest.fn()}/>);
    expect(tree).toMatchSnapshot();
  });
});