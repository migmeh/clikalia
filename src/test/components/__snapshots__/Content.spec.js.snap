// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Should render correctly 1`] = `
<div
  className="contx"
  id="content"
>
  <div
    className="dos cards-table"
  />
  <div
    className="s"
  />
</div>
`;
